lib/stdlib/test/property_test/lists_prop.erl

```diff
@@ -0,0 +1,1764 @@
+check_reversed(L1, L2) ->
+    check_reversed(L1, L2, []).
+
+check_reversed(L1, L2, Tail) ->
+    check_reversed1(L1, L2) =:= Tail.
+
+check_reversed1([], L2) ->
+    L2;
+check_reversed1([E|L1], L2) ->
+    case check_reversed1(L1, L2) of
+        [E|L2Rest] -> L2Rest;
+        _ -> false
+    end.
+
+%% --------------------------------------------------------------------
+check_seq([F|Seq], F, T, S) ->
+    check_seq(Seq, F + S, T, S);
+check_seq([], F, T, S) when S >= 0 ->
+    F >= T;
+check_seq([], F, T, S) when S < 0 ->
+    F =< T;
+check_seq(_Seq, _F, _T, _S) ->
+    false.
+
+%% --------------------------------------------------------------------
+check_sorted(L, Sorted) ->
+    check_sorted(fun erlang:'=<'/2, L, Sorted).
+
+check_sorted(SortFun, L, Sorted) ->
+    Elems = sets:from_list(L),
+    check_sorted1(SortFun, Sorted, Elems).
+
+check_sorted1(_SortFun, [], _Elems) ->
+    true;
+check_sorted1(SortFun, [E], Elems) ->
+    sets:is_element(E, Elems) andalso
+    check_sorted1(SortFun, [], Elems);
+check_sorted1(SortFun, [E1|L=[E2|_]], Elems) ->
+    SortFun(E1, E2) andalso
+    sets:is_element(E1, Elems) andalso
+    check_sorted1(SortFun, L, Elems);
+check_sorted1(_SortFun, _L, _Elems) ->
+    false.
+
+%% --------------------------------------------------------------------
+check_splitwithed(Pred, [E|L], [E|P1], P2) ->
+    Pred(E) andalso
+    check_splitwithed(Pred, L, P1, P2);
+check_splitwithed(Pred, L=[E|_], [], P2) ->
+    not Pred(E) andalso L =:= P2;
+check_splitwithed(_Pred, [], [], []) ->
+    true;
+check_splitwithed(_Pred, _L, _P1, _P2) ->
+    false.
+
+%% --------------------------------------------------------------------
+check_umerged(Ls, ML) ->
+    check_umerged(fun erlang:'=<'/2, Ls, ML).
+
+check_umerged(Fn, [[]|Ls], ML) ->
+    check_umerged(Fn, Ls, ML);
+check_umerged(_Fn, [L], ML) ->
+    ML =:= L;
+check_umerged(_Fn, [], ML) ->
+    ML =:= [];
+check_umerged(Fn, Ls, [E|ML]) ->
+    case find_remove_in_heads(Fn, E, Ls) of
+        {true, Ls1} ->
+            check_umerged(Fn, Ls1, ML);
+        false ->
+            false
+    end;
+check_umerged(_Fn, _Ls, _ML) ->
+    false.
+
+find_remove_in_heads(Fn, E, Ls) ->
+    find_remove_in_heads(false, Fn, E, Ls, []).
+
+find_remove_in_heads(Found, Fn, E, [[]|Ls], Seen) ->
+    find_remove_in_heads(Found, Fn, E, Ls, Seen);
+find_remove_in_heads(false, _Fn, _E, [], _Seen) ->
+    false;
+find_remove_in_heads(true, _Fn, _E, [], Seen) ->
+    {true, lists:reverse(Seen)};
+find_remove_in_heads(Found, Fn, E, [L=[E1|LRest]|Ls], Seen) ->
+    case Fn(E, E1) andalso Fn(E1, E) of
+        true ->
+            find_remove_in_heads(true, Fn, E, Ls, [LRest|Seen]);
+        false ->
+            find_remove_in_heads(Found, Fn, E, Ls, [L|Seen])
+    end.
+
+%% --------------------------------------------------------------------
+check_uniqed(L, UL) ->
+    check_uniqed(fun(X) -> X end, L, UL).
+
+check_uniqed(Fn, L, UL) ->
+    check_uniqed1(Fn, L, UL, sets:new()).
```

Comment:
```suggestion
    check_uniqed1(Fn, L, UL, sets:new([{version, 2}])).
```

Reply:
We were thinking of that, but our impression was that `sets:new/1` was supposed to be a temporary addition to the `sets` module for the time while there are several versions to choose from, which would disappear again afterwards?

Reply:
We have not decided yet when we will make version 2 default. There are potential backward compatibility problems if some application has saved a set to a disk file. Not sure if that is a problem in practice.

I suggest that you use `sets:new/1` for now. When we'll change the default, we will change all `sets:new/1` calls in the Erlang/OTP source code to `sets:new/0`.

Reply:
Ok, done :)

Btw, the docs for `sets:new/0` should probably say what version you will get from it. Right now, that can only be guessed from the paragraph at the top of the module description.